### PR TITLE
Instructions for Capacity Block for ML + EKS

### PR DESCRIPTION
**Pull request by sean-smith**
Capacity Blocks for ML have a [restriction](https://docs.aws.amazon.com/eks/latest/userguide/capacity-blocks.html) that prevents them from being created in a *managed node group*. This pull request adds a template that creates an unmanaged node group and adds support for EFA.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by verdimrc — APPROVED** [review]

File: 1.architectures/4.amazon-eks/README.md [hunk 1]
```diff
@@ -28,7 +28,8 @@ The following example cluster configurations for distributed training are provid
...
 * [**`eks-p4de-odcr.yaml`**](./eks-p4de-odcr.yaml): Cluster with 2 * `p4de.24xlarge` instances from an existing ODCR. A new VPC will be created for this cluster. This configuration is useful for distributed training when no VPC is already available. Note that you would have to match the AZ of your ODCR in the nodegroup section of the manifest. Nodegroups in this and previous examples are fully-managed and can be accessed via the EKS console. If you are using an instance type that is not yet supported in managed nodegroups by EKS, you can define a nodegroup in a self-manged nodegroup section as shown at the end of this example.
-
+* [**`eks-p5-odcr.yaml`**](./eks-p4de-odcr-vpc.yaml): Cluster with 1 * `p5.48xlarge` instances from an existing ODCR and an existing VPC. Note that you would have to match the AZ of your ODCR in the nodegroup section of the manifest. Nodegroups in this and previous examples are fully-managed and can be accessed via the EKS console. If you are using an instance type that is not yet supported in managed nodegroups by EKS, you can define a nodegroup in a self-manged nodegroup by using the `eks-p5-capacity-block.yaml` template.
```
Comment (perifaws) [inline 1.architectures/4.amazon-eks/README.md:31]: self-manged -> self-managed, typo. For the template, the bullet `eks-p5-odcr.yaml` links to `./eks-p4de-odcr-vpc.yaml`

File: 1.architectures/4.amazon-eks/README.md [hunk 2]
```diff
@@ -28,7 +28,8 @@ The following example cluster configurations for distributed training are provid
...
+* [**`eks-p5-capacity-block.yaml`**](./eks-p4de-odcr.yaml): This deploys a cluster without a node group, allowing you to create an unmanaged node group for Capacity Blocks for ML. See the section [Capacity Block](#4-capacity-blocks) for further detail.
```
Comment (perifaws) [inline 1.architectures/4.amazon-eks/README.md:32]: The bullet `eks-p5-capacity-block.yaml` links to `./eks-p4de-odcr.yaml`

Comment (sean-smith) [reply 1.architectures/4.amazon-eks/README.md:32]: resolved

File: 1.architectures/4.amazon-eks/README.md (hunk 1 above)
Comment (sean-smith) [reply 1.architectures/4.amazon-eks/README.md:31]: resolved

**Pull request merged**